code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -373,7 +373,7 @@
 
 		var/new_heat_capacity = air.heat_capacity()
 		if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
-			air.set_temperature(clamp(((old_thermal_energy + reaction_energy)/new_heat_capacity),TCMB,INFINITY))
+			air.set_temperature(clamp(((air.return_temperature()*old_heat_capacity + reaction_energy)/(new_heat_capacity*0.5)),TCMB,INFINITY))
```

Comment:
Why is there a `*0.5` here?

Reply:
because of auxmos, that 0.5 makes it works better